pkgs/by-name/gc/gcc-xtensa-lx106-embedded-8/package.nix

```diff
@@ -0,0 +1,64 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gcc-xtensa-lx106";
+  version = "2020r3";
+
+  suffix =
+    {
+      x86_64-linux = "linux-amd64";
+      x86_64-darwin = "mac";
+      i686-linux = "linux-i686";
+    }
+    .${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+
+  src = fetchurl {
+    url = "https://dl.espressif.com/dl/xtensa-lx106-elf-gcc8_4_0-esp-${version}-${suffix}.tar.gz";
+    hash =
+      {
+        x86_64-linux = "sha256-ChgEteIjHG24tyr2vCoPmltplM+6KZVtQSZREJ8T/n4=";
+        x86_64-darwin = "sha256-eCPF/fPOorWIzwxb2WJ3SrYKMLjZwZrgBWoO82l1N9s=";
+        i686-linux = "sha256-yNzn6OtYwf304DunTrgJToA6C1uK1qnVcSFvYOtyyzY=";
+      }
+        .${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+  };
+
+  phases = [ "unpackPhase" "installPhase" ];
```

Comment:
```suggestion
```
Explicit use of the `phases` is deprecated. Instead disable individual phases with `dontFoo`, e.g. `dontInstall = true;`

Reply:
I've changed it, but I'll push my commit after getting `autoPatchelfHook` working.

Reply:
I'm not sure why, but something goes wrong during autoPatchelfHook.
The hook AFAIK has two steps, 1 it removes unnecessary paths from the RDPATH of the binary and then 2 it patches the libraries in the binary. The second step seems to fail with `wrong ELF type`-error.

Build logs using autoPatchelfHook: https://pastebin.com/7x3x66Ga

I think manually patching the binaries is a good compromise to get it working, and using the autoPatchelfHook can be a fix later.

I've also seen a project creating wrappers using `buildFHSUserEnv` ([here](https://github.com/mirrexagon/nixpkgs-esp-dev/blob/master/pkgs/esp8266-rtos-sdk/esp8266-toolchain-bin.nix)), but I'm not sure if that is better or worse.